src/control/cmd/dmg/pretty/storage_query.go

```diff
@@ -44,19 +52,241 @@ func PrintHostStorageUsageMap(hsm control.HostStorageMap, out io.Writer) error {
 		hosts := getPrintHosts(hss.HostSet.RangedString())
 		row := txtfmt.TableRow{hostsTitle: hosts}
 		storage := hss.HostStorage
-		row[scmTitle] = humanize.Bytes(storage.ScmNamespaces.Total())
-		row[scmFreeTitle] = humanize.Bytes(storage.ScmNamespaces.Free())
-		row[scmUsageTitle] = storage.ScmNamespaces.PercentUsage()
-		row[nvmeTitle] = humanize.Bytes(storage.NvmeDevices.Total())
-		row[nvmeFreeTitle] = humanize.Bytes(storage.NvmeDevices.Free())
-		row[nvmeUsageTitle] = storage.NvmeDevices.PercentUsage()
+
+		sns := storage.ScmNamespaces
+		row[scmTitle] = humanize.Bytes(sns.Total())
+		scmFree := sns.Free()
+		row[scmFreeTitle] = humanize.Bytes(scmFree)
+		row[scmUsageTitle] = common.PercentageString(sns.Total()-scmFree, sns.Total())
+
+		ncs := storage.NvmeDevices
+		row[nvmeTitle] = humanize.Bytes(ncs.Total())
+		nvmeFree := ncs.Free()
+		row[nvmeFreeTitle] = humanize.Bytes(nvmeFree)
+		row[nvmeUsageTitle] = common.PercentageString(ncs.Total()-nvmeFree, ncs.Total())
+
 		table = append(table, row)
 	}
 
 	tablePrint.Format(table)
+	return
+}
+
+const (
+	metaRole  = storage.BdevRoleMeta
+	dataRole  = storage.BdevRoleData
+	rankTitle = "Rank"
+)
+
+// Return role combinations for each tier that contains either a meta or data role.
+func getTierRolesForHost(nvme storage.NvmeControllers, metaRolesLast, dataRolesLast *storage.BdevRoles) error {
+	roles := make(map[int]*storage.BdevRoles)
+	for _, c := range nvme {
+		if c.Roles().HasMeta() {
+			if _, exists := roles[metaRole]; !exists {
+				roles[metaRole] = c.Roles()
+			}
+		} else if c.Roles().HasData() {
```

Comment:
Probably missing something, but I do not understand why it is an else if and not an if.

Reply:
if meta and data exist on the same tier I don't add separate map entry for data, this avoids having to do an equality check later